lib/logstash/codecs/telemetry.rb

```diff
@@ -258,26 +345,75 @@ def ts_codec_extract_path_key_value(path, data, filter_table, depth)
     end # branch, v iteration over hash passed down
   end # ts_codec_extract_path_key_value
 
+  private
+  def decode_kv_table(table,evs)
```

Comment:
A more generous function name might work better here. E.g. produce_event_from_gpbkv_stream. A comment including an excerpt from the .proto might make it easier to follow too.